clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
@hachikuji I could give the approach of using a parameter shot if you want. I like the end result a lot better too?
On the flip side, it is a larger change than the approach here, which is fairly safe if I change the locking to `synchronized (this.wakeup)` in my opinion (we add additional locking that fixes the instability without softening locking elsewhere). Maybe there is value in going for the small change that adds `synchronized (this.wakeup)` and exposes `maybeTriggerWakeup` here and adding a JIRA for the move to a parameter (less risk + that change would become a lot more traceable)?